pkg/controller/applications/comp.go

```diff
@@ -29,3 +30,28 @@ func IsApplicationUpToDate(cr *v1alpha1.ApplicationParameters, remote *argocdv1a
 
 	return cmp.Equal(*cluster, remote.Spec, opts...) && maps.Equal(cr.Annotations, remote.Annotations) && slices.Equal(cr.Finalizers, remote.Finalizers)
 }
+
+// GetApplicationCondition evaluates the application status and returns appropriate Crossplane ready state
+func GetApplicationCondition(status *v1alpha1.ArgoApplicationStatus) xpv1.Condition {
```

Comment:
Nit: We don't need this to be exported.

```suggestion
// getApplicationCondition evaluates the application status and returns appropriate Crossplane ready state
func getApplicationCondition(status *v1alpha1.ArgoApplicationStatus) xpv1.Condition {
```